core/templates/dev/head/domain/dashboard/DashboardBackendApiService.js

```diff
@@ -33,7 +33,26 @@ oppia.factory('DashboardBackendApiService', [
     });
   };
 
+  var _fetchExplorationStats = function(explorationId) {
+    var requestUrl = (
+      '/dashboardhandler/explorationstats/?exp_id=' + explorationId);
+    return $q(function(resolve, reject) {
+      $http.get(requestUrl, {
+        cache: true
+      }).then(function(response) {
+        if (resolve) {
```

Comment:
Done. Also, reading a few parts of the code at https://github.com/angular/angular.js/blob/master/src/ng/http.js lead me to believe that angular's `$http` already uses `$q` internally!